docs/process-definition/branches.md

```diff
@@ -0,0 +1,95 @@
+[test infrastructure]: ../architecture.md#testing-infrastructure
+[release documentation]: releases.md
+[incubation]: incubation.md
+[bugs and small feature requests]: flow-standard.md
+[Dashboard Advisors]: advisors.md
+[tracker]: https://github.com/opendatahub-io/odh-dashboard/issues?q=is%3Aopen+is%3Aissue+label%3Atracker
+
+# Branches
+
+There are really two types of branches.
+
+- Fork Branch
+  - When you create a fork of our repo, these are the branches you push to your fork (often times called `origin`)
+  - These are often the source of PRs (with a few exceptions, see below)
+- Upstream Branch
+  - Feature branches
+  - Core branches (like `main` and `incubation`)
+  - [Bot branches](#bot-branches)
+
+Every _new_ commit needs to come from a Fork through a PR. We don't allow for pushing new content directly through our flows. New docs file, new code change, and even fixing a typo needs a PR from your fork to get into our repository.
+
+With that said, there are really 3 types of flows that utilize both Fork branches and Upstream branches.
+
+![branchesComplex.png](meta%2FbranchesComplex.png)
+
+Notes:
+
+- `main` this is our base branch, all code stems from here
+- `incubation` this is our "bleeding edge" of our feature set, this came from `main` but [cannot ever return to it](#never-returns-to-main)
+- Feature branches are the main other flow, they start from `main` and will eventually return to `main` - often times with a stay in `incubation`
+
+There is only ever 1 `main` and 1 `incubation` branch. Feature branches start with `f/`, and can be as many as needed.
+
+Read more on git tags & releases in our [release documentation].
+
+## Main
+
+> aka "The Stable Branch"
+
+`main` is our most stable and tested code. This flow should be as stable as we are today. We look to only push [bugs and small feature requests] into this branch. Everything else goes to [feature branches](#feature-branches), so it can make it into `incubation` and get closer to stability before merging into `main`.
+
+When a feature branch merges into `main` it can be officially considered _done_ and up to our standards for stable.
+
+## Incubation
+
+> Note: This branch is an amalgamation of several feature branches, source code here is often struggling to co-exist among the features. Conflicts are normal merging into it.
```

Comment:
nit: last sentence is clunky, it could perhaps read:

"Conflicts are normal *when* merging into it."